packages/contracts-bedrock/test/setup/ForkLive.s.sol

```diff
@@ -97,20 +101,53 @@ contract ForkLive is Deployer {
 
         // Now deploy the updated OPCM and implementations of the contracts
         _deployNewImplementations();
+
+        // Now upgrade the contracts
+        _upgrade();
     }
 
     /// @notice Etches a new Deploy.s.sol contract at a deterministic address, sets up the environment,
     ///         and deploys new implementations with a "_NextVersion" suffix. This suffix is necessary to avoid
     ///         naming collisions with the implementations saved above.
     function _deployNewImplementations() internal {
+        // Etch a copy of the Deploy.s.sol contract so that we can use its deployImplementations() function
         Deploy deployNew = Deploy(address(uint160(uint256(keccak256(abi.encode("optimism.deploy.new"))))));
         vm.etch(address(deployNew), vm.getDeployedCode("Deploy.s.sol:Deploy"));
         vm.label(address(deployNew), "DeployNew");
         vm.allowCheatcodes(address(deployNew));
-        vm.setEnv("CONTRACT_ADDRESSES_PATH", string.concat(vm.projectRoot(), "/deployments/1-deploy.json"));
 
+        // Set the CONTRACT_ADDRESSES_PATH environment variable
+        // This is used in Deploy.setUp() to read and save the addresses it needs (SuperchainConfigProxy,
+        // ProtocolVersionsProxy)
+        string memory jsonPath = string.concat(vm.projectRoot(), "/deployments/1-deploy.json");
+        vm.setEnv("CONTRACT_ADDRESSES_PATH", jsonPath);
         deployNew.setUp();
         deployNew.deployImplementations({ _isInterop: false, _suffix: "_NextVersion" });
+
+        // deployNew.deployImplementations() saves the addresses to its own storage as well as to the json file at
+        // CONTRACT_ADDRESSES_PATH. In order to use the addresses in the json file, we now need to load them into
+        // local storage.
+        loadAddresses(jsonPath);
+    }
+
+    function _upgrade() internal {
+        OPContractsManager opcm = OPContractsManager(mustGetAddress("OPContractsManager_NextVersion"));
+
+        ISystemConfig systemConfig = ISystemConfig(mustGetAddress("SystemConfigProxy"));
+        IProxyAdmin proxyAdmin = IProxyAdmin(EIP1967Helper.getAdmin(address(systemConfig)));
+
+        address upgrader = proxyAdmin.owner();
+        vm.label(upgrader, "ProxyAdmin Owner");
+
+        ISystemConfig[] memory systemConfigs = new ISystemConfig[](1);
+        systemConfigs[0] = systemConfig;
+        IProxyAdmin[] memory proxyAdmins = new IProxyAdmin[](1);
+        proxyAdmins[0] = proxyAdmin;
```

Comment:
Consider adding helpers for this to our vendored `Solarray.sol`, or if you want to go further we can fork the [original repo](https://github.com/emo-eth/solarray) since improved array UX for these would be useful in superchain-ops too. Can of course be tracked in a separate issue

Reply:
Unfortunately, this test itself will continue to only do a single upgrade upgrade. We don't have a plan and it would be pretty complicated to setup testing of multiple forked networks in a single call.